spec/front_desk_spec.rb

```diff
@@ -0,0 +1,398 @@
+        @concierge = build_front_desk
+      end # before
+
+      it "will return array of available rooms" do
+        reservation_3 = @concierge.make_reservation(
+          1001,
+          3,
+          15,
+          Date.new(2019, 4, 8),
+          Date.new(2019, 4, 21),
+        )
+        @concierge.add_reservation(reservation_3)
+
+        all_available_rooms = @concierge.room_availability(
+          Date.new(2019, 4, 8)
+        )
+
+        expect(all_available_rooms).must_be_kind_of Array
+        expect(all_available_rooms.length).must_equal 17
+      end # it
+    end # describe
+
+    describe "raise an ArgumentError if check_in date is before current date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an error if check_in date is before current date" do
+        expect {
+          @concierge.make_reservation(
+            30011,
+            1,
+            19,
+            Date.new(2019, 3, 10),
+            Date.new(2019, 3, 25)
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "raise an ArgumentError if check_out date is before check_in date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an error if check_out date is before check_in date" do
+        expect {
+          @concierge.make_reservation(
+            30111,
+            1,
+            20,
+            Date.new(2019, 3, 28),
+            Date.new(2019, 3, 25)
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "raise an ArgumentError if check_out date is before current date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an ArgumentError if check_out date is before current date" do
+        expect {
+          @concierge.make_reservation(
+            30111,
+            1,
+            20,
+            Date.new(2019, 9, 8),
+            Date.new(2019, 2, 25)
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "block cannot have more than 5 rooms" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will throw an error if there is an attempt to create a block with more than 5 rooms" do
+        expect {
+          @concierge.make_reservation(
+            450111,
+            7,
+            10,
+            Date.new(2019, 8, 8),
+            Date.new(2019, 8, 12)
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "Number of rooms in Hotel cannot exceed 20" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will throw an error if there is an attempt to create a reservation for a room beyond 20" do
+        expect {
+          @concierge.make_reservation(
+            5501,
+            5,
+            17,
+            Date.new(2019, 10, 8),
+            Date.new(2019, 10, 12)
+          )
+        }.must_raise ArgumentError
```

Comment:
It isn't clear from this test why this is relevant to the test (it looks like a regular, valid reservation)-- maybe there are ways to use variable names and error detail testing to make this clear